app/parsers/bulkrax/bagit_parser.rb

```diff
@@ -134,7 +134,8 @@ def write_files
 
         record.file_sets.each do |fs|
           file_name = filename(fs)
-          next if file_name.blank?
+          next if file_name.blank? || fs.original_file.blank?
```

Comment:
if we exit the loop here, the FileSet does not get written to the csv.

Reply:
When this happens and the FileSet does not get written to the CSV, then because of this line https://github.com/samvera-labs/bulkrax/blob/f81734b00f9e7aa91301ac39e23a748986b5d252/app/parsers/bulkrax/bagit_parser.rb#L127
it leads me to think it still would be counted resulting in a split that's instead of 1000 records, something like 999 records.

Would a `records_in_folder - 1` or something to that extend be appropriate here?

Reply:
maybe something like?

```ruby
if file_name.blank? || fs.original_file.blank?
  records_in_folder - 1
  next
end
```

Reply:
great suggestion! thanks!

Reply:
Yeah, i was looking at that... didn't know how to get around that 😓

Reply:
I considered weeding out the bad files after the `#current_record_ids` method, and before the `#write_files` methods. that would require finding each file set and then removing the id if `original_file` doesn't exist though.

in this case where we are exporting the entire tenant, there are over 2000 files. plus `current_record_ids` isn't an instance variable and we call it a few times so it could result in different values. I don't think the operational cost of removing a few missing files so that we can have 1000 records exactly (or however many they deem once that value becomes dynamic) on a csv is worth it.

Reply:
Agreed, I gave it some thought as well and I don't think it'll be worth it either.